hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterClockImpl.java

```diff
@@ -46,4 +47,16 @@ public void setMasterTime(long masterTime) {
     public long getClusterTimeDiff() {
         return (clusterTimeDiff == Long.MAX_VALUE) ? 0 : clusterTimeDiff;
     }
+
+    @Override
+    public long getClusterStartTime() {
```

Comment:
Thanks but please see this bug report and test the provided reproducer code there with the older java versions than 8 https://bugs.openjdk.java.net/browse/JDK-6523160, according to my tests and bug report page, it is already fixed in java 8. If any java version returns a negative value for uptime, it will be a bug. So, as far as i understand, there is a possibility that all nodes may return negative value for node uptimes in this PR. To me this should be fixed, but at least possible anomalies should be documented if it will not be fixed. 
Because a negative uptime value may be surprising.